File: QA-project/backend/app/models/reports.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from app.database import Base


class Report(Base):
    __tablename__ = "reports"

    report_id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.user_id", ondelete="CASCADE"), nullable=False)
    dataset_id = Column(Integer, ForeignKey("datasets.dataset_id", ondelete="CASCADE"), nullable=False)
    report_type = Column(String(100), nullable=False)
    file_path = Column(String(500))
    generated_at = Column(DateTime, server_default=func.now(), nullable=False)

    # Relationships
    user = relationship("User", back_populates="reports")
    dataset = relationship("Dataset", back_populates="reports")

File: QA-project/backend/src/db/schema/reports.ts
import {
  pgTable,
  serial,
  integer,
  varchar,
  timestamp,
} from "drizzle-orm/pg-core";
import { relations } from "drizzle-orm";
import { users } from "./users";
import { datasets } from "./datasets";

export const reports = pgTable("reports", {
  reportId: serial("report_id").primaryKey(),
  userId: integer("user_id")
    .notNull()
    .references(() => users.userId, { onDelete: "cascade" }),
  datasetId: integer("dataset_id")
    .notNull()
    .references(() => datasets.datasetId, { onDelete: "cascade" }),
  reportType: varchar("report_type", { length: 100 }).notNull(),
  filePath: varchar("file_path", { length: 500 }),
  generatedAt: timestamp("generated_at").defaultNow().notNull(),
});

export const reportsRelations = relations(reports, ({ one }) => ({
  user: one(users, {
    fields: [reports.userId],
    references: [users.userId],
  }),
  dataset: one(datasets, {
    fields: [reports.datasetId],
    references: [datasets.datasetId],
  }),
}));
